tables/tabulate-molecular-alterations.R

```diff
@@ -0,0 +1,396 @@
+               "TACC1", "TCF4", "TERT", "TP53", "YAP1", "ZIC1")
+
+# Later we'll need to use a list to fill in NAs with tidyr::replace_na()
+gene_fill_list <- list()
+for (gene in gene_list) gene_fill_list[[gene]] <-  "None"
+
+## Set up paths ----------------------------------------------------------------
+
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+data_dir <- file.path(root_dir, "data")
+palette_file <- file.path(root_dir,
+                          "figures",
+                          "palettes",
+                          "broad_histology_cancer_group_palette.tsv")
+metadata_file <- file.path(data_dir, "pbta-histologies.tsv")
+maf_file <- file.path(data_dir,
+                      "pbta-snv-consensus-mutation.maf.tsv.gz")
+hotspots_maf_file <- file.path(data_dir,
+                               "pbta-snv-scavenged-hotspots.maf.tsv.gz")
+cnv_autosomes_file <- file.path(data_dir,
+                                "consensus_seg_annotated_cn_autosomes.tsv.gz")
+cnv_xy_file <- file.path(data_dir,
+                         "consensus_seg_annotated_cn_x_and_y.tsv.gz")
+fusion_file <- file.path(data_dir,
+                         "pbta-fusion-putative-oncogenic.tsv")
+
+# Output CSV file
+zenodo_csv_file <- file.path(root_dir,
+                             "tables",
+                             "zenodo-upload",
+                             "openpbta-molecular-alterations.csv")
+
+## Read in data ----------------------------------------------------------------
+
+# metadata
+histologies_df <- readr::read_tsv(metadata_file, guess_max = 10000)
+palette_df <- readr::read_tsv(palette_file)
+
+# maf and hotspots data
+maf_df <- readr::read_tsv(maf_file)
+hotspot_df <- readr::read_tsv(hotspots_maf_file)
+
+# fusion data
+fusion_df <- readr::read_tsv(fusion_file)
+
+# CNV data: autosomes and XY
+cnv_df <- readr::read_tsv(cnv_autosomes_file) %>%
+  dplyr::bind_rows(
+    readr::read_tsv(cnv_xy_file)
+  )
+
+## Prepare the maf data --------------------------------------------------------
+
+# Only keep maf columns relevant for annotating specific alterations
+maf_keep_cols <-c("Tumor_Sample_Barcode",
+                  "Hugo_Symbol",
+                  "Variant_Type",
+                  "HGVSp",
+                  "HGVSc")
+
+# subset columns first to enable combining dfs; they have some different
+#  data types in other columns we're getting rid of
+hotspot_df <- hotspot_df %>%
+  dplyr::select(maf_keep_cols)
+
+# Final MAF alterations
+maf_df <- maf_df %>%
+  dplyr::select(maf_keep_cols) %>%
+  # combine consensus MAF and hotspot MAF
+  dplyr::bind_rows(hotspot_df) %>%
+  # filter to only relevant genes and ids
+  dplyr::filter(Hugo_Symbol %in% gene_list) %>%
+  dplyr::distinct() %>%
+  dplyr::mutate(
+    # assign the final alteration based on which piece
+    # of information is known, in this order of priority
+    final_alteration = dplyr::case_when(
+      HGVSp != "." ~ HGVSp,
+      HGVSc != "." ~ HGVSc,
```

Comment:
Agree that short form makes sense here, but we probably also want to use the nucleotide change if there is no change in the protein, which I am doing here by grabbing the last character, which will be `=` if there is no change. 
```suggestion
      HGVSp_Short != "."  & 
        stringr::str_sub(HGVSp_Short, -1) != "=" ~ HGVSp_Short,
      HGVSc != "." ~ HGVSc,
```